src/main/ApplicationUtils.cpp

```diff
@@ -387,31 +387,34 @@ catchup(Application::pointer app, CatchupConfiguration cc,
     auto synced = false;
     asio::io_context::work mainWork(io);
     auto done = false;
+    auto& cm = app->getCatchupManager();
     while (!done && clock.crank(true))
     {
-        switch (app->getLedgerManager().getState())
+        if (!cm.isCatchupInitialized())
```

Comment:
I don't think this is needed: catchup has to be initialized with `app->getLedgerManager().startCatchup(cc, archive);` on line 370

Reply:
done